File: array/181.cpp
#include <bits/stdc++.h>

using namespace std;


int dominantIndex(vector<int>& nums) {
    vector<int>::iterator maxElement = max_element(nums.begin(), nums.end());
    int index = distance(nums.begin(), maxElement);

    for(int i = 0; i < nums.size(); ++i) {
        if(nums[i] == *maxElement)
            continue;
        if(*maxElement < nums[i] * 2)
            return -1;
    }

    return index;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);


    vector<int> nums {1,2,3,4};
    cout<<boolalpha<<dominantIndex(nums);


    return 0;
}

File: array/181.py
from typing import List


def dominantIndex(nums: List[int]) -> int:
    largest, second = 0, 0
    index_largest_num = 0

    for index, value in enumerate(nums):
        if value > largest:
            second = largest
            largest = value
            index_largest_num = index
        elif value > second:
            second = value

    if largest >= second * 2:
        return index_largest_num
    else:
        return -1


print(dominantIndex([1, 2, 3, 4]))
